raitools/raitools/_managers/explainer_manager.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Explainer Manager class."""
+
+import warnings
+from scipy.sparse import issparse
+from interpret_community.mimic.mimic_explainer import MimicExplainer
+from interpret_community.mimic.models.lightgbm_model import (
+    LGBMExplainableModel)
+from interpret_community.mimic.models.linear_model import (
+    LinearExplainableModel)
+from interpret_community.common.constants import ModelTask
+from raitools._managers.base_manager import BaseManager
+
+SparseNumFeaturesThreshold = 1000
+
+
+class ExplainerManager(BaseManager):
+
+    """Defines the ExplainerManager for explaining a model.
+
+    :param model: The model to explain.
+        A model that implements sklearn.predict or sklearn.predict_proba
+        or function that accepts a 2d ndarray.
+    :type model: object
+    :param initialization_examples: A matrix of feature vector
+        examples (# examples x # features) for initializing the explainer.
+    :type initialization_examples: pandas.DataFrame
+    :param evaluation_examples: A matrix of feature vector
+        examples (# examples x # features) on which to explain the
+        model's output.
+    :type evaluation_examples: pandas.DataFrame
+    :param target_column: The name of the label column.
+    :type target_column: str
+    :param classes: Class names as a list of strings.
+        The order of the class names should match that of the model
+        output.  Only required if explaining classifier.
+    :type classes: list
+    """
+
+    def __init__(self, model, initialization_examples, evaluation_examples,
+                 target_column, classes=None):
+        """Defines the ExplainerManager for explaining a model.
+
+        :param model: The model to explain.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param initialization_examples: A matrix of feature vector
+            examples (# examples x # features) for initializing the explainer.
+        :type initialization_examples: pandas.DataFrame
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param classes: Class names as a list of strings.
+            The order of the class names should match that of the model
+            output.  Only required if explaining classifier.
+        :type classes: list
+        """
+        self._model = model
+        self._initialization_examples = \
```

Comment:
I have a feeling that all the managers will be doing something like this... might want to refactor

Reply:
doesn't fairness use the whole dataset?

Reply:
That's actually something which popped up in the latest update to the AzureML API proposal. The `add()` method for the `FairnessManager` has acquired a `dataset` argument which can be `initialisation` `evaluation` or `both` (which would concat the two first)

Reply:
Also, are these going to be copies or references? We don't want to be keeping multiple copies of the data around.

Reply:
references, but removing datasets on the explanations means we need to make changes in interpret-community, I think that's out of scope for this PR but it might be something we can do long-term

Reply:
You don't need to make changes to `interpret-community`. All you need to do is have the `get()` method reinsert them before returning the explanation object to the user.

Reply:
Sorry I'm confused.  Why would we need to reinsert them if we want to get rid of them from the explanation?  And if we are ok with keeping them on the explanation, and they are references already, why would we want to reinsert them?

Reply:
Will they still be references after we've saved to disk and reloaded?

If the user wants an explanation, then we would need to reinsert the datasets. However, if they're then firing up the ModelAssessmentDashboard, won't that know how to get the common datasets whenever required, rather than needing them embedded in the explanations themselves? Again, the the transition to the TS layer, will the data remain referenced, or would we end up with multiple copies?

Reply:
I think you've highlighted multiple issues here, but that code hasn't been implemented -- when it will be implemented, we should make sure to use references and not make copies.

Reply:
But why not build it in from the beginning?

We certainly could make it so that each component is a literal copy of what we have in the individual dashboards now. However, that then means we have technical debt going forward, in removing the duplication. And if we've written files like that, then things are particularly painful since we'll have to keep supporting the older files.

Reply:
I wrote a new comment on teams.  Maybe we can allow users to return explanations without datasets on them.  There was already a user request for this:
https://github.com/interpretml/interpret-community/issues/368
I can add this as a new feature in interpret-community.